mypyc/emitmodule.py

```diff
@@ -216,62 +304,151 @@ def generate_c_for_modules(self) -> List[Tuple[str, str]]:
             else:
                 decls.emit_lines(*declaration.decl)
 
-        if self.shared_lib_name:
+        if self.group_name:
+            self.generate_export_table(ext_declarations, emitter)
+
             self.generate_shared_lib_init(emitter)
 
         ext_declarations.emit_line('#endif')
         declarations.emit_line('#endif')
 
-        return file_contents + [('__native.c', ''.join(emitter.fragments)),
-                                ('__native_internal.h', ''.join(declarations.fragments)),
-                                ('__native.h', ''.join(ext_declarations.fragments)),
+        return file_contents + [('__native{}.c'.format(self.group_suffix),
+                                 ''.join(emitter.fragments)),
+                                ('__native_internal{}.h'.format(self.group_suffix),
+                                 ''.join(declarations.fragments)),
+                                ('__native{}.h'.format(self.group_suffix),
+                                 ''.join(ext_declarations.fragments)),
                                 ]
 
+    def generate_export_table(self, decl_emitter: Emitter, code_emitter: Emitter) -> None:
+        """Generate the declaration and definition of the group's export struct.
+
+        To avoid needing to deal with deeply platform specific issues
+        involving dynamic library linking (and some possibly
+        insurmountable issues involving cyclic dependencies), compiled
+        code accesses functions and data in other compilation groups
+        via an explicit "export struct".
+
+        Each group declares a struct type that contains a pointer to
+        every function and static variable it exports. It then
+        populates this struct and stores a pointer to it in a capsule
+        stored as an attribute named 'exports' on the group's shared
+        library's python module.
+
+        On load, a group's init function will import all of its
+        dependencies' exports tables using the capsule mechanism and
+        copy the contents into a local copy of the table (to eliminate
+        the need for a pointer indirection when accessing it).
+
+        Then, all calls to functions in another group and accesses to statics
+        from another group are done indirectly via the export table.
```

Comment:
Maybe give a concrete example of what the export table looks like, and another group can access and item from it (using C fragments)?